reV/supply_curve/point_summary.py

```diff
@@ -428,6 +379,153 @@ def mean_h5_dsets_data(self):
+        ----------
+        data : ndarray
+            data layer vector to compute mode for
+
+        Returns
+        -------
+        float | int
+            Mode of data
+        """
+        return stats.mode(data).mode[0]
+
+    @staticmethod
+    def _categorize(data, excl_mult):
+        """
+        Extract the sum of inclusion scalar values (where 1 is
+        included, 0 is excluded, and 0.7 is included with 70 percent of
+        available land) for each unique (categorical value) in data
+
+        Parameters
+        ----------
+        data : ndarray
+            Vector of categorical values
+        excl_mult : ndarray
+            Vector of inclusion values
+
+        Returns
+        -------
+        str
+            Jsonified string of the dictionary mapping categorical values to
+            total inclusions
+        """
+        data = {category: float(excl_mult[(data == category)].sum())
+                for category in np.unique(data)}
+        data = jsonify_dict(data)
+
+        return data
+
+    @classmethod
+    def _agg_data_layer_method(cls, data, excl_mult, method):
+        """Aggregate the data array using specified method.
+
+        Parameters
+        ----------
+        data : np.ndarray | None
+            Data array that will be flattened and operated on using method.
+            This must be the included data. Exclusions should be applied
+            before this method.
+        excl_mult : np.ndarray | None
+            Scalar exclusion data for methods with exclusion-weighted
+            aggregation methods. Shape must match input data.
+        method : str
+            Aggregation method (mode, mean, max, min, sum, category)
+
+        Returns
+        -------
+        data : float | int | str | None
+            Result of applying method to data.
+        """
+        method_func = {'mode': cls._mode,
+                       'mean': np.mean,
+                       'max': np.max,
+                       'min': np.min,
+                       'sum': np.sum,
+                       'category': cls._categorize}
+
+        if data is not None:
+            method = method.lower()
+            if method not in method_func:
+                e = ('Cannot recognize data layer agg method: '
+                     '"{}". Can only {}'.format(method, list(method_func)))
+                logger.error(e)
+                raise ValueError(e)
+
+            if len(data.shape) > 1:
+                data = data.flatten()
+
+            if data.shape != excl_mult.shape:
+                e = ('Cannot aggregate data with shape that doesnt '
+                     'match excl mult!')
+                logger.error(e)
+                raise DataShapeError(e)
+
+            if method == 'category':
+                data = method_func['category'](data, excl_mult)
+            elif method == 'mode':
+                data = method_func['mode'](data)
+            else:
+                data = data * excl_mult
```

Comment:
Did you see our thread on teams? I dont think (max, min, mode) should be weighted. I do think that (mean, sum, categorize) should be weighted.

Reply:
Missed that, will update